test/photometry.jl

```diff
@@ -98,3 +106,71 @@ end
     @test table.aperture_sum[1] ≈ 25π
     @test all(table.aperture_sum[2:end] .< 25π)
 end
+
+#################################################
+#Test for elliptical apertures
+
+@testset "outside - $AP" for (AP, params) in zip(APERTURES1, PARAMS1)
+    data = ones(10, 10)
+    aperture = AP(-60, 60, params...)
+    @test aperture_photometry(aperture, data, method = (:subpixel, 10)).aperture_sum ≈ 0
+end
+
+@testset "inside zeros - $AP" for (AP, params) in zip(APERTURES1, PARAMS1)
+    data = zeros(40, 40)
+    aperture = AP(20.0, 20.0, params...)
+
+    table_cent = aperture_photometry(aperture, data, method = :center)
+    table_sub = aperture_photometry(aperture, data, method = (:subpixel, 10))
+
+    @test table_sub.aperture_sum ≈ 0
+    @test table_cent.aperture_sum ≈ 0
+
+end
+
+@testset "inside ones - $AP" for (AP, params) in zip(APERTURES1, PARAMS1)
+    data = ones(40, 40)
+    aperture = AP(20.0, 20.0, params...)
+
+    table_cent = aperture_photometry(aperture, data, method = :center)
+    table_sub = aperture_photometry(aperture, data, method = (:subpixel, 10))
+
+    true_flux = area(aperture)
+
+    @test table_sub.aperture_sum ≈ true_flux atol = 1
+    @test table_cent.aperture_sum ≈ true_flux atol = 1
+
+end
+
+function test_elliptical_aperture(data, aperture)
+    error = ones(size(data))
+
+    table_cent = aperture_photometry(aperture, data, error, method = :center)
+    table_sub = aperture_photometry(aperture, data, error, method = (:subpixel, 12))
+
+    true_flux = area(aperture)
+    true_err = sqrt(true_flux)
+
+    @test table_sub.aperture_sum ≈ true_flux atol = 1
+    @test table_cent.aperture_sum ≈ true_flux atol = 5
+    @test table_sub.aperture_sum_err ≈ true_err atol = 1
+    @test table_cent.aperture_sum_err ≈ true_err atol = 1
```

Comment:
The absolute tolerance on these is _very_ high. I wouldn't trust a method that is only accurate to +- 5 values of flux. This suggests to me there may be something weird happening with the calculations. I would do some side-by-side comparisons to photutils (remember their positioning starts at 0, though) to see if you are significantly different than their algorithm.

Reply:
In general I don't like using absolute tolerance, it doesn't tell me anything about the significance of the error

Reply:
Okay, I will definitely look into it. :slightly_smiling_face: